jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -241,6 +245,22 @@ static boolean hasSetOutEncoding() {
 	static PrintStream createConsole(FileDescriptor desc, Charset charset) {
 		BufferedOutputStream bufStream = new BufferedOutputStream(new FileOutputStream(desc));
 		Charset consoleCharset = charset == null ? consoleDefaultCharset : charset;
+
+		/*[IF JAVA_SPEC_VERSION >= 19]*/
+		Properties props = internalGetProperties();
+		if (FileDescriptor.out == desc) {
+			if ((null != setOutProp) || (null == props.getProperty("stdout.encoding"))) { //$NON-NLS-1$
+				setOutProp = consoleCharset.name();
```

Comment:
I don't see how this could be called twice (for the same `FileDescriptor`).
And even if that were possible, why should we modify the system property?

Reply:
This should never overwrite existing values for `stdout.encoding` nor `stderr.encoding`. I suggest:
```
		/*[IF JAVA_SPEC_VERSION >= 19]*/
		Properties props = internalGetProperties();
		if (FileDescriptor.out == desc) {
			props.putIfAbsent("stdout.encoding", consoleCharset.name()); //$NON-NLS-1$
		} else if (FileDescriptor.err == desc) {
			props.put("stderr.encoding", consoleCharset.name()); //$NON-NLS-1$
		}
		/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
```

Reply:
We do want to overwrite stdout.encoding, and stderr.encoding in one case.
At startup when createConsole() is originally called, only the java.base module is loaded, so we only have access to the encodings in java.base. The platform may be using an encoding from jdk.charsets (or the user may be requesting one). We set stdout.encoding and stderr.encoding to ASCII for the duration of startup. Later from J9VMInternals, createConsole() is called again once the encoding can be resolved from jdk.charsets, and then we overwrite the stdout.encoding and stderr.encoding with the final value.